### DISCUSSION TITLE
Multi-channel status by channel near real time.

### DISCUSSION THREAD

**Discussion by KS4EC** [rtl-airband/RTLSDR-Airband#430] [Ideas]
So... Since I am a nerd that likes to know what is going on, I built a display with python that shows what chan's are in use almost real time.
I did this with the help of many of you by turning on record to file and separate on transmission, Sorry, memory issues so that might not be the exact names.  (See discussion #306)

I then wrote a python script to watch for those files in a ram drive and when they had the extension .tmp I knew the channel was active and when they changed to .mp3 I knew that the activity had ended and deleted the mp3 file.

I used an SQLite DB to store the status as real time as it could, and used another python script that monitored the DB to refresh the display as things changed.

It actually works quite well, but I was wondering if something could be added to the software to write to a DB right from Airband, instead of writing and deleting files.  Maybe a config option for SQL, or CSV if people don't want to use SQL. Updated at least 1x per sec.

Anyway, this opens up an option for a multichannel console "Radio" for monitoring all chan's at once and setting what chan's are priority vs background or muted while always seeing channel activity. Nice for busy airports.

Any thoughts on this? Is it easy to do, or just to hard and probably not used often?

Thanks for all you do and creating such a great SDR platform.

Rob - KS4EC

**Comment by charlie-foxtrot** [comment]
Looking for `.tmp` vs `.mp3` files is a very creative solution!

Writing to a DB seems a bit to far from the "core" of with the software does, however it may make sense to have a way to stream stats / metadata that is more detailed than the `.prom` files every 15 seconds.  Some way to indicate squelch open / closed per channel via a socket or other stream perhaps?

This also reminds me of #324

**Comment by KS4EC** [comment]
I get it, but just having the option to write to sqlite3 like specify db and table name.Let the user set up the sql if they want it.
Other thoughts were a 1 line csv file with boolean fields.But that may have locking problems.  I tried that before and the locks drive me nuts trying to read and write at the same time.
Maybe simple environment variables that a nother script could monitor, name them with the Chan tag or freq?
Just some thoughts.
I will help where I can as I am now retired due to medical issues but I am not a real good programmer.  Just learning python.
Rob

Sent from AT&T Yahoo Mail on Android 
 
  On Wed, Oct 25, 2023 at 12:23 PM, ***@***.***> wrote:   


Looking for .tmp vs .mp3 files is a very creative solution!

Writing to a DB seems a bit to far from the "core" of with the software does, however it may make sense to have a way to stream stats / metadata that is more detailed than the .prom files every 15 seconds. Some way to indicate squelch open / closed per channel via a socket or other stream perhaps?

This also reminds me of #324
